src/Data/Array/Polarized/Pull.hs

```diff
@@ -0,0 +1,54 @@
+{-# OPTIONS_GHC -fno-warn-partial-type-signatures #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE NoImplicitPrelude #-}
+{-# LANGUAGE PartialTypeSignatures #-}
+
+-- | This module is intended to be imported qualified, e.g.
+-- > import qualified Data.Array.Polarized.Pull as Pull
+module Data.Array.Polarized.Pull
+  ( Array
+  , zip, zipWith
+  , append
+  , make
+  , foldr
+  , foldMap
+  , findLength
+  , asList
+  , singleton
+  , fromFunction
+  , fromVector
+  , toVector
```

Comment:
We're lacking a `split` as well. I see splitting functions as the defining characteristic of pull arrays.